src/modules/user/services/user.service.ts

```diff
@@ -50,6 +51,9 @@ class UserService extends BaseMongoService<UserDto> {
       return this.throwError("User already exists", StatusConflict);
     }
 
+    if (data.password) {
+      data.password = await bcrypt.hash(data.password, 10);
+    }
```

Comment:
```suggestion
    data.password = await bcrypt.hash(data.password, 10);
```
unnecessary if check, handled by http layer (yup validator)

Reply:
i ll update to this `data.password = await bcrypt.hash(data.password as string, 10);` cause in yup the type is string | undefined

Reply:
`const isValidPassword = await bcrypt.compare(data.password!, user.password);`
@Rendy752